### PR TITLE
[60] Mobile Navbar

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- Refactor old navbar
- Implement mobile navbar
- Add `useBackNavbar()`  for this type of header 
image
must insert a value that wrapped in useMemo
```ts
useBackNavbar(useMemo(()=>({title:"ค้นหางาน", onBack:()=>{}}),[]))
```
- Add `useNavbarSearch()` for setting the onClick function of search icon. The function must be wrapped in useCallback (search icon won't appear if you don't call this)
image
```ts
useNavbarSearch(useCallback(()=>{},[]))
```  
- Mobile profile is still dummy (next PR)
- Will write test in the next PR (because I think that some branch is waiting for this feature)

## Demo

- https://dev.modela.miello.dev/

### CODE REVIEW

File: apps/web/src/common/components/Navbar/components/NavbarProfile/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      <img
+        src="https://genshin.honeyhunterworld.com/img/ayaka_002.webp"
+        alt="profile image"
+        style={{
+          width: '40px',
+          height: '40px',
+          objectFit: 'cover',
+          borderRadius: '50%',
+        }}
+      />
```
Comment (miello) [inline apps/web/src/common/components/Navbar/components/NavbarProfile/index.tsx:19]: Next image ?

File: apps/web/src/common/components/Navbar/components/NavbarProfile/styled.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import styled from '@emotion/styled'
```
Comment (miello) [inline apps/web/src/common/components/Navbar/components/NavbarProfile/styled.ts:1]:
```suggestion
import styled from '@mui/material'
```

File: apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    if (!user)
+      return VERIFIED_MENU.filter(
+        (item) => item.allowNotLoggedIn || item.onlyNotLoggedIn,
+      )
+    if (!user.isVerified)
+      return VERIFIED_MENU.filter(
+        (item) => item.allowNotVerified && !item.onlyNotLoggedIn,
+      )
+    return VERIFIED_MENU.filter((item) => !item.onlyNotLoggedIn)
+  }
```
Comment (miello) [inline apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx:80]: useCallback ?

File: apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  const getMenuByRole = () => {
+    if (user?.type !== UserType.CASTING)
+      return getMenuByUser().filter((item) => !item.castingOnly)
+    return getMenuByUser()
+  }
```
Comment (miello) [inline apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx:86]: useCallback ?

File: apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+  const getMenuByScreenSize = (isMobile: boolean) => {
+    if (isMobile) return getMenuByRole().filter((item) => !item.desktopOnly)
+    return getMenuByRole().filter((item) => !item.mobileOnly)
+  }
```
Comment (miello) [inline apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx:91]: useCallback

File: apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      allowNotVerified: true,
+    },
+    {
+      label: 'เข้าสู่ระบบ',
+      onClick: () => router.push('/login'),
+      icon: <LoginOutlined />,
+      allowNotVerified: true,
+      onlyNotLoggedIn: true,
+    },
+  ]
```
Comment (miello) [inline apps/web/src/common/components/Navbar/hooks/useNavMenu/index.tsx:68]: Does we need useMemo here ? (not sure)

File: apps/web/src/common/context/LayoutContext/types.ts [hunk 7]
```diff
@@ -3,6 +3,20 @@ export interface ILayoutContext {
   setNavbarFocus: (focus: NavbarFocus) => void
   isHideNavbar: boolean
   setHideNavbar: (isHideNavbar: boolean) => void
+  onSearch: (() => void) | null
+  setOnSearch: (callback: (() => void) | null) => void
+  override: NavbarOverride | null
+  setOverride: (override: NavbarOverride | null) => void
```
Comment (miello) [inline apps/web/src/common/context/LayoutContext/types.ts:9]: (Optional)
Actual type of `set` is `React.Dispatch<SetStateAction<T>>`

File: apps/web/src/common/hooks/useBackNavbar/index.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { NavbarOverride } from 'common/context/LayoutContext/types'
+import { useEffect } from 'react'
+
+import { useLayout } from '../../context/LayoutContext'
```
Comment (miello) [inline apps/web/src/common/hooks/useBackNavbar/index.ts:4]:
```suggestion
import { useLayout } from 'common/context/LayoutContext'
```

**Review by miello — CHANGES_REQUESTED** [review]

File: apps/web/src/common/components/Navbar/components/NavbarProfile/index.tsx (hunk 1 above)
Comment (PKhing) [reply apps/web/src/common/components/Navbar/components/NavbarProfile/index.tsx:19]: will do this when fetch user data from API

**Comment by miello** [comment]
In mobile size, I think logout button should have some space to bottom of the window
image

**Comment by miello** [comment]
At mobile size, hover the icon on top right does not show the cursor icon (It's not serious when using on mobile or ipad). I think we should add it just in case

**Review by miello — APPROVED** [review]

**Pull request merged**